### ISSUE TITLE
Username in Windows file path with apostrophe throws error

### ISSUE THREAD

**Issue by disicinco15** [sveltejs/kit#3669]
### Describe the bug

When trying using the command "npm run dev" from the VS Code terminal, the following error below (in reproduction) is given.  It stems from the username (C:\Users\[User Name]\... has an apostrophe in the string.

### Reproduction

PS C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb> npm run dev

> moviedb@0.0.1 dev
> svelte-kit dev

 > html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/generated/root.svelte:1:38: error: Expected ";" but found "s"
    1 │ import 'virtual-module:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/generated/root.svelte';
      ╵                                       ^

 > html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/runtime/components/error.svelte:9:38: error: Expected ";" but found "s"
    9 │ import 'virtual-module:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/runtime/components/error.svelte';
      ╵                                       ^

> Build failed with 2 errors:
html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/generated/root.svelte:1:38: error: Expected ";" but found "s"
html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/runtime/components/error.svelte:9:38: error: Expected ";" but found "s"
html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/generated/root.svelte:1:38: error: Expected ";" but found "s"
html:C:/Users/Chuck's Mac/Documents/Professional/General Web Dev learning/Svelte Movie DB/moviedb/.svelte-kit/runtime/components/error.svelte:9:38: error: Expected ";" but found "s"
    at failureErrorWithLog (C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:1493:15)        
    at C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:1151:28
    at runOnEndCallbacks (C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:941:63)
    at buildResponseToResult (C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:1149:7)       
    at C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:1258:14
    at C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:629:9
    at handleIncomingPacket (C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:726:9)
    at Socket.readFromStdout (C:\Users\Chuck's Mac\Documents\Professional\General Web Dev learning\Svelte Movie DB\moviedb\node_modules\esbuild\lib\main.js:596:7)        
    at Socket.emit (node:events:520:28)
    at addChunk (node:internal/streams/readable:324:12)

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (8) x64 Intel(R) Core(TM) i7-8565U CPU @ 1.80GHz
    Memory: 8.78 GB / 15.82 GB
  Binaries:
    Node: 17.4.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 97.0.4692.99
    Edge: Spartan (44.22000.120.0), Chromium (97.0.1072.76)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    svelte: ^3.44.0 => 3.46.3
```

### Severity

blocking all usage of svelte

**Comment by Conduitry** [comment]
This isn't exclusive to Windows - I can also reproduce it with a path containing a `'` on Linux. The code that throws the error that I end up with is things like `import 'virtual-module:/.../'/.svelte-kit/generated/root.svelte';`, so I'm not sure whether this is something on Vite's end or on our end.

**Comment by Conduitry** [comment]
I suspect this is a Vite issue.

https://github.com/vitejs/vite/blob/5a111cedf31f579e3b8c8af5c4442d2e0cd5aa12/packages/vite/src/node/optimizer/scan.ts#L287

This should probably be something like `import ${JSON.stringify(virtualModulePrefix + path);\n`.

**Comment by Rich-Harris** [comment]
From an initial exploration I _think_ it's a Vite issue, though I don't have a repro handy.

**Comment by Conduitry** [comment]
Opened https://github.com/vitejs/vite/issues/6717

**Comment by bluwy** [comment]
Fixed in Vite 2.8.1, which can be updated by running `npm update`.
